stdlib/LinearAlgebra/src/eigen.jl

```diff
@@ -685,6 +685,16 @@ function show(io::IO, mime::MIME{Symbol("text/plain")}, F::Union{Eigen,Generaliz
     nothing
 end
 
+function Base.hash(F::Eigen, h::UInt)
+    return hash(F.values, hash(F.vectors, h))
```

Comment:
Should this hash the type of the factorization as well? I kept it consistent with the previous behavior, but perhaps that should be changed?

Reply:
I have changed this now to include `Eigen` in the hash.